### DISCUSSION TITLE
浅谈本方案的各种用法

### DISCUSSION THREAD

**Discussion by ksqsf** [rimeinn/rime-moran#40] [General]
其实，和许多用户一样，我自己也是首次接触此类单字辅码连打的方案，因此在制作之外，也在探索这样的方案到底怎么用效率才高。这里简单写一点探索，抛砖引玉……（汗，搞得好像有人看一样。）

# 基础用法：纯双拼，整句模式和默认模式皆可

在绝大多数情况下，本方案可以当作纯双拼方案使用。

# 基础用法：整句模式

如果依旧比较习惯传统的双拼输入方式，但需要减少选字量，则使用不带简码的整句模式，仅在需要选重时加辅筛选即可。

（此外，本方案还含有一个辅筛模式，也许有人在用？）

# 基础用法：字词模式

传统四码方案。适合需要精准录入的场景（如跟打），日常聊天也比较爽。熟习默认模式的简码后可以随开随用，无需另行学习。

# 以短语为单位输入：默认模式

上面的用法都是一句话就能解释清楚的，下面的则复杂一点儿。

和默认模式比较接近的常规输入法形态是四码方案挂接智能拼音，相比于挂接用法，默认模式一方面更强（可以加词辅），另一方面又更弱（不改 Lua 的话，打不了四码“简码”）。

我自己的用法现在稳定在以「短语」为单位输入。这里的短语指一个语义单元（下文简称「语素」），我会自由地利用不定长方案的灵活性，跨越四码的边界，如：

> 一个人的一生/假如是/丰富多彩/的: yigerfdeyiug_jwruui_fgfudocl_d_
> 一个人/的/一生/假如是/丰富多彩/的: ygr_d_yiug_jwruui_fgfudocl_d_
> 
> PS: 我对简码很熟，很多词语实际输入时会下意识地拆单，但这不意味着我认为它们是分离的语素：`一个人/的/一/生/假如/是/丰富多彩/的` ygr_d_y_ug_jwru_u_fgfudocl_d_

我只要觉得一个语言要素是连贯在一起的，就会把它们放在一起打，打完之后按一下空格确认上屏。这是我个人认为最适合日常打字的用法，也是我觉得最适合我自己的用法。

对于不同的语素，可以采用不同的打法取得更好的效果：

1. 对于单字，有简打简，无简打o
2. 对于二字词，有把握就直接按空格，把握不大就加一位辅码（同样，有简打简）
3. 对于三字以上词，重码本就不多，无需加辅，直接打全即可（同样，有简打简）

这种打法的优势在于兼顾了效率和确定性，对普通现代文本来说有较好的效果：

1. 码长适中：根据一些实测，多数实际语句的码长在 2.3~2.5 之间，接近普通形码打词水平。
2. 具有一定确定性：简码本就是固定的，而词加辅后多数情况也可以直接离散，仅少数情况需要选重（受到自动调频影响）
3. 不费脑：以音为主，不用动脑；需要用形码的字和词不多，容易形成条反，然后就可以肆意击键 :-D （日常打到5、6击不是难事，虽然想文字的速度一般没有那么快……）

从某种意义上说，是回归了传统的双拼加形辅的用法。但是与传统的形辅方案（和魔然的辅筛方案）不同，默认模式的注意力焦点在句末而不是句首，从体验上来说是边打边选字的过程，而不是打完之后统一选字，这一点造成了不小的使用体验差异。当然，我更倾向于本方案的用法：关注的焦点越局域化，要处理的信息量就越少，处理速度就越快。

# 三码连打

另外，由于可以整句加辅，还有一个有趣的用法是可以三码连打——所有字都打三码，然后自动组句。

<img width="397" alt="image" src="https://github.com/ksqsf/rime-moran/assets/23358293/dbe79986-0da7-4088-9f72-a9b862407d9a">

（可通过修改方案实现每个字都是三码的效果，避免拆分歧义。即使不修改效果也不错。）

据少量实测，这种打法即使总是打整句，确定性也挺高。当然，坏处就是每个字都要加辅，不管码长还是击键都没有优势，速度显然没有上一节的用法更快，或许不宜日用？

# 其他更多用法？

虚位以待……

**Comment by liangkaichun** [comment]
我是20年的自然双拼用户，双拼输入的节奏感不太好改变，所以大多数时候都是纯双拼输入，遇到词库中没有的词或者单字，会增加辅码来辅筛和造词。

我用了魔然默认的方案和词库，但手动去掉了英文、日文、各种反查，以及闪电字词，保留了双拼+辅码的能力，以及tab键+辅码造词的能力。我自己维护了一个仅保留修改后魔然默认方案和词库的配置文件，然后只更新词库。

魔然方案最吸引我的就是可以辅码造词，而不仅是辅码筛选。这个能力当前做的比较好的是落格输入法，遇到词库中没有的词，可以直接输入首字辅码、次字辅码，选字造词。
